feat(app): #4 Adding Survival Mode

rizkyalviandra opened the pull request:
add survival mode in invgame

rizkyalviandra committed:
feat(app): 4 Adding Survival Mode

add survival mode in invgame

ronaudinho commented on cmd/survival/main.go:
```diff
@@ -0,0 +1,275 @@
+// else overlap is fine
+func gen(prev incantate) (bool, incantate, []string) {
+	// var cast bool
+	cast := true
```
`cast` can be false/true in this mode

rizkyalviandra replied:
I already make the `cast` can be true or false

ronaudinho commented on cmd/survival/main.go:
```diff
@@ -0,0 +1,275 @@
+		m.timer, cmd = m.timer.Update(msg)
+		return m, cmd
+	case timer.TimeoutMsg:
+		m.isQuit = true
```
`isQuit` doesn't seem necessary as this quits anyway

ronaudinho replied:
also, printing the total time taken before quitting is needed in this mode

rizkyalviandra replied:
I already printed the total time taken before quitting the mode, but I don't think that same as in invokergame

ronaudinho commented on cmd/survival/main.go:
```diff
@@ -0,0 +1,275 @@
+			if _, ok := m.incantate[m.invoked[0]]; ok {
+				delete(m.incantate, m.invoked[0])
+				m.point++
+				m.timer.Timeout += 2 * time.Second
```
not privy to how it actually works in invokergame, but seems like each set of spells have assigned timeout, instead of adding to existing time. looks like the logic belongs in `gen`? ditto for the treatment in `case "f"`below

rizkyalviandra replied:
same as above, i made the timeout reset to 10 seconds when spell invoked or cast, but i don't think that same as in invokergame

ronaudinho commented on cmd/survival/main.go:
```diff
@@ -0,0 +1,275 @@
+			// a spell casting.
+			if _, ok := m.incantate[m.invoked[0]]; ok {
+				delete(m.incantate, m.invoked[0])
+				m.point++
```
actually i dont think we care about points in survival

rizkyalviandra replied:
I have already taken out all point methods from this mode

ronaudinho commented on cmd/survival/main.go:
```diff
@@ -0,0 +1,275 @@
+	}
+	s += "\n--------------------------------------------------\n"
+
+	s += fmt.Sprintf("%d POINTS\n\n", m.point)
```
same as above, points is not relevant in this mode

rizkyalviandra replied:
same as above

rizkyalviandra committed:
fix(app): #4 fixing in game method on survival mode

takeout point on survival mode and fixing how to add timeout when spell invoked or cast

Closes: #4

ronaudinho reviewed (approved):
some improvements on timeout we can make in other PR, merging.

@RiskyFeryansyahP feel free to ~roast~ leave comments when you have the time.
